cmd/dendrite-monolith-server/main.go

```diff
@@ -61,22 +74,38 @@ func main() {
 	rsAPI := roomserver.SetupRoomServerComponent(
 		base, keyRing, federation,
 	)
+	if base.EnableHTTPAPIs {
+		rsAPI = base.CreateHTTPRoomserverAPIs()
+	}
+
 	eduInputAPI := eduserver.SetupEDUServerComponent(
 		base, cache.New(),
 	)
+	if base.EnableHTTPAPIs {
+		eduInputAPI = base.CreateHTTPEDUServerAPIs()
+	}
+
 	asAPI := appservice.SetupAppServiceAPIComponent(
 		base, accountDB, deviceDB, federation, rsAPI, transactions.New(),
 	)
+	if base.EnableHTTPAPIs {
+		asAPI = base.CreateHTTPAppServiceAPIs()
+	}
+
 	fsAPI := federationsender.SetupFederationSenderComponent(
 		base, federation, rsAPI, &keyRing,
 	)
+	if base.EnableHTTPAPIs {
+		fsAPI = base.CreateHTTPFederationSenderAPIs()
```

Comment:
Uhh can we batch this please? E.g
```go
if base.EnableHTTPAPIs {
    eduInputAPI = base.CreateHTTPEDUServerAPIs()
    asAPI = base.CreateHTTPAppServiceAPIs()
    fsAPI = base.CreateHTTPFederationSenderAPIs()
}
```

Reply:
They are used as parameters when setting up other components so the ordering is important.

Reply:
Oh :( sadness